tests/baselines/reference/genericMappedTypeAcceptsInferredObjectType.errors.txt

```diff
@@ -0,0 +1,72 @@
+
+
+==== tests/cases/compiler/genericMappedTypeAcceptsInferredObjectType.ts (7 errors) ====
+    interface Constraint {
+    	foo: string
+    }
+    
+    interface ExtendedConstraint extends Constraint {
+    	bar: string
+    }
+    
+    type GenericMap<T> = {
+    	[P in keyof T]: T[P]
+    }
+    
+    type OptionalGenericMap<T> = {
+    	[P in keyof T]?: T[P]
+    }
+    
+    const required = <T>(x: GenericMap<T>) => x
+    const optional = <T>(x: OptionalGenericMap<T>) => x
+    
+    const withinConstraint = <T extends Constraint>(foo: T['foo']) => {
+    	required<Constraint>({ foo }) // no error as { foo: T['foo'] } <: GenericMap<Constraint>
+    	required<T>({ foo }) // error as { foo: T['foo'] } /<: GenericMap<T> because other properties may be missing
+    	            ~~~~~~~
+!!! error TS2345: Argument of type '{ foo: T["foo"]; }' is not assignable to parameter of type 'GenericMap<T>'.
+    	optional<T>({}) // no error as {} <: OptionalGenericMap<Constraint>
+    	optional<T>({ foo }) // no error as { foo: T['foo'] } <: OptionalGenericMap<T>
+    }
+    
+    const withinExtendedConstraint = <T extends ExtendedConstraint>(foo: T['foo'], bar: T['bar']) => {
+    	required<ExtendedConstraint>({ foo }) // error as { foo: T['foo'] } /<: GenericMap<ExtendedConstraint> because bar is missing
+    	                             ~~~~~~~
+!!! error TS2345: Argument of type '{ foo: T["foo"]; }' is not assignable to parameter of type 'GenericMap<ExtendedConstraint>'.
+!!! error TS2345:   Property 'bar' is missing in type '{ foo: T["foo"]; }' but required in type 'GenericMap<ExtendedConstraint>'.
+!!! related TS2728 tests/cases/compiler/genericMappedTypeAcceptsInferredObjectType.ts:6:2: 'bar' is declared here.
+    	required<ExtendedConstraint>({ bar }) // error as { bar: T['bar'] } /<: GenericMap<ExtendedConstraint> because foo is missing
+    	                             ~~~~~~~
+!!! error TS2345: Argument of type '{ bar: T["bar"]; }' is not assignable to parameter of type 'GenericMap<ExtendedConstraint>'.
+!!! error TS2345:   Property 'foo' is missing in type '{ bar: T["bar"]; }' but required in type 'GenericMap<ExtendedConstraint>'.
+!!! related TS2728 tests/cases/compiler/genericMappedTypeAcceptsInferredObjectType.ts:2:2: 'foo' is declared here.
+    	required<ExtendedConstraint>({ foo, bar }) // no error as { foo: T['foo'], bar: T['bar'] } <: GenericMap<ExtendedConstraint>
+    	required<T>({ foo, bar }) // error as { foo: T['foo'], bar: T['bar'] } /<: GenericMap<T> because other properties may be missing
+    	            ~~~~~~~~~~~~
+!!! error TS2345: Argument of type '{ foo: T["foo"]; bar: T["bar"]; }' is not assignable to parameter of type 'GenericMap<T>'.
+    	optional<T>({}) // no error as {} <: OptionalGenericMap<T>
+    	optional<T>({ foo }) // no error as { foo: T['foo'] } <: OptionalGenericMap<T>

```

Comment:
Maybe another hint for the reviewers because the threads might look a little convoluted: I would consider this line not erroring a _nice to have_ but not a necessity to fix the referenced issue. So it might be best to leave this unaddressed and improve this behavior further in a follow-up. 🙂 

Here are my thoughts on this matter up to this point: https://github.com/microsoft/TypeScript/pull/42382#discussion_r566676535